src/main/java/com/qaprosoft/carina/demo/gui/pages/AutomationPracticeSearchResultPage.java

```diff
@@ -21,11 +21,31 @@ public AutomationPracticeSearchResultPage(WebDriver driver) {
     }
 
     public AutomationPracticeAddedProductModalPage addProductToCart(String name) {
-        for (ProductCard product:products) {
-            if(product.getProductName().equalsIgnoreCase(name))
-                product.clickAddToCartButton();
+        ProductCard productCard;
+        for (int i = 0; i < products.size(); i++) {
+            productCard = products.get(i);
+            if(productCard.getProductName().equalsIgnoreCase(name)) {
+                productCard.clickAddToCartButton();
+            }
         }
         return new AutomationPracticeAddedProductModalPage(driver);
     }
+
+    public String getResultsCounter(){
+        return resultsCounter.getText();
+    }
+
+    public boolean isAnyResultFound() {
+        return !resultsCounter.getText().equalsIgnoreCase("\n" +
+                "            0 results have been found.        ");
+    }
+
+    public boolean isAnyProductShown() {
+        return !products.isEmpty();
+    }
+
+    public boolean resultAndCounterMatch() {
+        return products.size() == Integer.parseInt(resultsCounter.getText().replaceAll("[^0-9]", ""));
```

Comment:
i suggest not to verify it in page object, this verification will look better in test. So, you need here method getProductQuantity()